cirq-core/cirq/transformers/analytical_decompositions/quantum_shannon_decomposition_test.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cirq.transformers.analytical_decompositions.quantum_shannon_decomposition import (
+    _multiplexed_cossin,
+    _nth_gray,
+    _msb_demuxer,
+    _single_qubit_decomposition,
+    quantum_shannon_decomposition,
+)
+
+import pytest
+import numpy as np
+from scipy.stats import unitary_group
+import cirq
+
+
+@pytest.mark.parametrize('n_qubits', [_ for _ in range(1, 8)])
+def test_random_qsd_n_qubit(n_qubits):
+    U = unitary_group.rvs(2**n_qubits)
```

Comment:
You can also use `cirq.testing.random_unitary` to a get random unitary of a given dimension.